### ISSUE TITLE
How to use multi-output to get higher accuracy?

### ISSUE THREAD

**Issue by dzyjjpy** [CoinCheung/BiSeNet#75]
Dear old iron:
   There are three output node,  all three output nodes' shape are 1*Class_number*H*W.
From the code:
`out = self.net(im)[0]`
just use the first output node, no use for [1] and [2].
Is there any method that can improve the accuracy with multi-outputs?
@CoinCheung

**Comment by CoinCheung** [comment]
Hi, 

You can try to sum up the logits of all the outputs before using softmax to compute the score, though I do not think this would bring good result easily, since the auxiliary outputs are not rich in information of the extracted features. You can also try to sum up the softmax scores of all the outputs before using `torch.argmax` to compute the pred. I have not tried it, you can doing some experiements and see if it will work.

**Comment by dzyjjpy** [comment]
Got it. I will try it.  I guess maybe different weights for the three layers will be helpful.
Update after experiments

**Comment by CoinCheung** [comment]
Hi, 

Personally, I believe the aux outputs will all have some performance loss, since they have small spatial area. The primary output feature has a downsample rate of 8, while the auxiliary output features have downsample rate of 16 and 32. Their output may be too coarse to offer enough help. If you would like to have a try, I think you can try to add the auxiliary feature with downsample rate of 16 and discard the 32 times downsampled feature. Good luck.
